nautobot/dcim/models/device_components.py

```diff
@@ -758,6 +761,150 @@ def parent(self):
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+    status = StatusField(blank=False, null=False)
+    # Preemptively model 2.0 behavior by making `created` a DateTimeField rather than a DateField.
+    created = models.DateTimeField(auto_now_add=True)
+    description = models.CharField(
+        max_length=200,
+        blank=True,
+    )
+    interfaces = models.ManyToManyField(
+        to="dcim.Interface",
+        through="dcim.InterfaceRedundancyGroupAssociation",
+        related_name="interface_redundancy_groups",
+        blank=True,
+    )
+    protocol = models.CharField(
+        max_length=50,
+        blank=True,
+        choices=InterfaceRedundancyGroupProtocolChoices,
+        verbose_name="Redundancy Protocol",
+    )
+    protocol_group_id = models.CharField(
+        max_length=50,
+        blank=True,
+    )
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        default=None,
+        blank=True,
+        null=True,
+    )
+    virtual_ip = models.ForeignKey(
+        to="ipam.IPAddress",
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+        related_name="interface_redundancy_groups",
+    )
+
+    class Meta:
+        """Meta class."""
+
+        ordering = ["name"]
+
+    csv_headers = [
+        "name",
+        "status",
+        "description",
+        "protocol",
+        "protocol_group_id",
+        "secrets_group",
+        "virtual_ip",
+    ]
+
+    def to_csv(self):
+        return (
+            self.name,
+            self.get_status_display(),
+            self.description,
+            self.protocol,
+            self.protocol_group_id,
+            self.secrets_group.name if self.secrets_group else None,
+            str(self.virtual_ip) if self.virtual_ip else None,
+        )
+
+    def __str__(self):
+        """Return a string representation of the instance."""
+        return self.name
+
+    def add_interface(self, interface, priority):
+        """
+        Add an interface including `priority`.
+
+        :param interface:
+            Interface instance
+        :param priority:
+            Integer priority used by redundancy protocol
+        """
+        instance = self.interfaces.through(
+            interface_redundancy_group=self,
+            interface=interface,
+            priority=priority,
+        )
+        return instance.validated_save()
+
+    def remove_interface(self, interface):
+        """
+        Remove an interface.
+
+        :param interface:
+            Interface instance
+        """
+        instance = self.interfaces.through.objects.get(
+            interface_redundancy_group=self,
+            interface=interface,
+        )
+        return instance.delete()
+
+
+@extras_features(
+    "relationships",
+    "custom_fields",
+)
+class InterfaceRedundancyGroupAssociation(BaseModel, ChangeLoggedModel):
```

Comment:
Not blocking for this PR but something we may need to address promptly in 1.6.x before shipping 2.0rc1 --- `@extras_features(relationships, custom_fields)` on a `BaseModel` looks wrong to me, if this model is meant to support relationships and customfields then it needs to inherit from `RelationshipModel` and `CustomFieldModel`, which will likely be another database migration. A quicker/easier fix may be to remove the `extras_features` from the model if it doesn't actually need to support these properties.